quiz/src/test/java/rx/quiz/collections/ProStreamsKoanTest.java

```diff
@@ -0,0 +1,23 @@
+package rx.quiz.collections;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ProStreamsKoanTest extends ProStreamsKoan {
+
+    @NotNull
+    @Override
+    public String sortStringWithMultipleCriteria(@NotNull String input) {
+        return Stream.of(input.toLowerCase().split(" "))
+                .map(s -> s.chars().mapToObj(c -> String.valueOf((char) c)))
+                .map(s -> s.distinct().collect(Collectors.joining()))
+                .sorted((s1, s2) -> {
+                    if (s1.length() != s2.length()) {
```

Comment:
Good idea with if! We have used java.util.Comparator#thenComparing(java.util.Comparator<? super T>) or something from Guava

Reply:
Thanks. It's more readable with thenComparing